builder/dockerfile/builder.go

```diff
@@ -196,6 +199,11 @@ func (b *Builder) dispatchDockerfileWithCancellation(dockerfile *parser.Result)
 		// TODO: get this from dispatch
 		imageID = b.image
 
+		if b.Aux != nil {
+			if err := b.Aux.Emit(types.BuildResult{ID: imageID}); err != nil {
+				return "", err
+			}
+		}
```

Comment:
> The final image you can just handle after dispatchers have completed.

The downside of that is that the last message will now differ and require many test case updates (like my first version had). I think that's the right thing to do though, especially given that I now handle `DOCKER_API_VERSION` correctly. I'll make this change.